controllers/cloud.redhat.com/makers/makers_test.go

```diff
@@ -216,3 +220,173 @@ func TestMultiDependency(t *testing.T) {
 		t.Errorf("Wrong number of dep services")
 	}
 }
+
+type Params map[string]map[string]string
+type IDAndParams struct {
+	Params Params
+	ID     string
+}
+
+func NewIDAndParam(ID, limitCPU, limitMemory, requestsCPU, requestsMemory string) IDAndParams {
+	return IDAndParams{
+		ID: ID,
+		Params: Params{
+			"limits": {
+				"cpu":    limitCPU,
+				"memory": limitMemory,
+			},
+			"requests": {
+				"cpu":    requestsCPU,
+				"memory": requestsMemory,
+			},
+		},
+	}
+}
+
+func createResourceRequirements(params Params) core.ResourceRequirements {
+	rl := core.ResourceList{}
+	limits := params["limits"]
+
+	if limits != nil {
+		if limits["cpu"] != "" {
+			rl["cpu"] = resource.MustParse(limits["cpu"])
+		}
+		if limits["memory"] != "" {
+			rl["memory"] = resource.MustParse(limits["memory"])
+		}
+	}
+
+	rr := core.ResourceList{}
+	requests := params["requests"]
+	if requests != nil {
+		if requests["cpu"] != "" {
+			rr["cpu"] = resource.MustParse(requests["cpu"])
+		}
+		if requests["memory"] != "" {
+			rr["memory"] = resource.MustParse(requests["memory"])
+		}
+	}
+
+	return core.ResourceRequirements{
+		Limits:   rl,
+		Requests: rr,
+	}
+}
+
+func setupResourcesForTest(params Params) (*apps.Deployment, *crd.ClowdEnvironment, *core.ResourceRequirements) {
+	var app crd.ClowdApp
+
+	objMeta := metav1.ObjectMeta{
+		Name:      "reqapp",
+		Namespace: "default",
+		Labels: map[string]string{
+			"app": "test",
+		},
+	}
+
+	appResources := createResourceRequirements(params)
+
+	app = crd.ClowdApp{
+		ObjectMeta: objMeta,
+		Spec: crd.ClowdAppSpec{
+			Dependencies: []string{
+				"bopper",
+				"snapper",
+			},
+			Pods: []crd.PodSpec{{
+				Name:      "reqapp",
+				Resources: appResources,
+			}},
+		},
+	}
+
+	envResourceParams := Params{
+		"limits": {
+			"cpu":    "300m",
+			"memory": "1024Mi",
+		},
+		"requests": {
+			"cpu":    "30m",
+			"memory": "512Mi",
+		},
+	}
+
+	env := crd.ClowdEnvironment{
+		Spec: crd.ClowdEnvironmentSpec{
+			ResourceDefaults: createResourceRequirements(envResourceParams),
+		},
+	}
+
+	d := &apps.Deployment{}
+
+	nn := types.NamespacedName{
+		Name:      app.Spec.Pods[0].Name,
+		Namespace: app.Namespace,
+	}
+
+	initDeployment(&app, &env, d, nn, app.Spec.Pods[0], "hihi")
```

Comment:
I kind of feel like this function call should be in the calling function since this is the actual function we're testing.  Like, `setupResourcesForTest` just sets up the resources for running the test, where "the test" is `initDeployment`.